src/btree/bt_sync_obsolete.c

```diff
@@ -12,17 +12,18 @@
 #define WT_URI_FILE_PREFIX "file:"
 
 /*
- * __sync_obsolete_inmem_evict --
- *     Check whether the inmem ref is obsolete according to the newest stop time point and mark it
- *     for urgent eviction.
+ * __sync_obsolete_inmem_evict_or_mark_dirty --
+ *     This function checks whether the in-memory ref contains obsolete information and takes
+ *     necessary action.
  */
 static int
-__sync_obsolete_inmem_evict(WT_SESSION_IMPL *session, WT_REF *ref)
+__sync_obsolete_inmem_evict_or_mark_dirty(WT_SESSION_IMPL *session, WT_REF *ref)
```

Comment:
This function is called in a loop from `__checkpoint_cleanup_obsolete_cleanup`, should we check the cache state in the loop as well? For now it's only checked at the top level in `__checkpoint_cleanup_page_skip`. We could create a function that checks the cache state periodically.

Reply:
No. With this new change, this function is no longer called from `__checkpoint_cleanup_obsolete_cleanup`. We do not need to check the cache state in other places.